src/core/Crust/Plugins/hooks.ts

```diff
@@ -0,0 +1,232 @@
+import { useCallback, useEffect, useMemo } from "react";
+
+import useClientStorage from "@reearth/components/molecules/Visualizer/useClientStorage";
+import { CameraPosition } from "@reearth/core/mantle/compat/types";
+import { MouseEventHandles, MouseEvents } from "@reearth/core/Map";
+import events from "@reearth/core/Map/types/event";
+import { useGet } from "@reearth/core/Map/utils";
+
+import { commonReearth } from "./api";
+import { ReearthEventType, Viewport, ViewportSize } from "./plugin_types";
+import { Context, Props } from "./types";
+import usePluginInstances from "./usePluginInstances";
+
+export type SelectedReearthEventType = Pick<
+  ReearthEventType,
+  "cameramove" | "select" | "tick" | "resize" | keyof MouseEvents | "layeredit"
+>;
+
+export default function ({
```

Comment:
I did refactor to gather code related to Plugin. [`useProvider` defined in Visualizer](https://github.com/reearth/reearth-web/blob/ff47c54a53cdf165425a522625f255c9247abd26/src/components/molecules/Visualizer/hooks.ts#L483) is only used for plugin, so I moved this code to this function.
Also I ported [some code defined in `Visualizer/Plugin/context.tsx`](https://github.com/reearth/reearth-web/blob/ff47c54a53cdf165425a522625f255c9247abd26/src/components/molecules/Visualizer/Plugin/context.tsx#L164) to this function.